### PR TITLE
Documentation changes

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Some minor changes to our docs
- Add troubleshooting guide for grants/permissions config drifts
- Update authentication details in index.md

## Tests

- [ ] ~`make test` run locally~
- [x] relevant change in `docs/` folder
- [ ] ~covered with integration tests in `internal/acceptance`~
- [ ] ~relevant acceptance tests are passing~
- [ ] ~using Go SDK~

### CODE REVIEW

File: docs/guides/troubleshooting.md [hunk 1]
```diff
@@ -126,3 +126,29 @@ This error may appear when creating workspace-level objects but the provider is
 ### Error: Provider registry.terraform.io/databricks/databricks v... does not have a package available for your current platform, windows_386
 
 This kind of errors happens when the 32-bit version of Databricks Terraform provider is used, usually on Microsoft Windows.  To fix the issue you need to switch to use of the 64-bit versions of Terraform and Databricks Terraform provider.
+
+### Permanent configuration drifts with `databricks_grants` or `databricks_permissions`
+
+For both resources, each single instance should manages all the grants/permissions for a single object, etc. If there are multiple instances set up against an object, they will keep overwriting one another and lead to permanent configuration drifts.
```
[the remark this answers is not in the document]
Comment (alexott) [reply docs/guides/troubleshooting.md:132]: Maybe change it to

```
For both resources, each single resource instance should manage all the grants/permissions for a given object, etc. If there are multiple resource instances set up against the same object, ... 
```

?

File: docs/guides/troubleshooting.md [hunk 2]
```diff
@@ -126,3 +126,29 @@ This error may appear when creating workspace-level objects but the provider is
...
+
+To prevent that, you need to have only one resource, and inside that resource, use [Dynamic Blocks](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) to specify the variable number of nested grant blocks.
```
Comment (alexott) [inline docs/guides/troubleshooting.md:134]: `one resource` -> `one resource instance` ?

**Review by alexott — CHANGES_REQUESTED** [review]
small improvements would be nice to do

**Review by alexott — APPROVED** [review]
lgtm

**Review by mgyucht — APPROVED** [review]

**Pull request merged**